tests/test_patterns.py

```diff
@@ -41,14 +44,22 @@ def test_pattern_from_file_sequence():
     assert list(fp.items()) == list(zip(expected_keys, file_sequence))
 
 
-def test_file_pattern_concat_merge():
+@pytest.mark.parametrize("pickle", [False, True])
+def test_file_pattern_concat_merge(pickle):
     concat = ConcatDim(name="time", keys=list(range(3)))
     merge = MergeDim(name="variable", keys=["foo", "bar"])
 
     def format_function(time, variable):
         return f"T_{time}_V_{variable}"
 
     fp = FilePattern(format_function, merge, concat)
+
+    if pickle:
+        # regular pickle doesn't work here because it can't pickle format_function
+        from cloudpickle import dumps, loads
+
+        fp = loads(dumps(fp))
```

Comment:
Generally speaking, what is pickling (or, cloudpickling, in this case?) achieving here? Pickling is one of those concepts I've seen around the Python world a lot, but haven't understood yet. (Google exists, of course, but thought it might be worth asking within our specific context.)

Reply:
Why don't you move your comment to #117, which is specifically about this.